### ISSUE TITLE
disable security configuration app when there is no security api access

### ISSUE THREAD

**Issue by tremmert** [opendistro-for-elasticsearch/security-kibana-plugin#91]
At the moment, when a user does not have the right roles to use the security configuration the security configuration app is still visible, with some or all icons disabled..

Which is weird, misleading and just doesn't look very good...

![64494490-7233ee00-d285-11e9-813f-7f3477b7d31b](https://user-images.githubusercontent.com/1215797/65321734-e4b59f80-db9c-11e9-8c67-b36040998a1a.png)

I would suggest to completely disable the security config when people do not have the right access rights... 

The easiest way to do this, and I think the most consistent one is to disable it if there is no rest-api access, which is exposed through: `systemstate.getRestApiInfo().has_api_access`

Can be done in: 

`public/chrome/configuration/enable_configuration.js`

with this change:

`     // rest module installed, check if user has access to the API
     systemstate.loadRestInfo().then(function(){
-        chrome.getNavLinkById("security-configuration").hidden = false;
-        FeatureCatalogueRegistryProvider.register(() => {
-            return {
-                id: 'security-configuration',
-                title: 'Security Configuration',
-                description: 'Configure users, roles and permissions for Open Distro Security.',
-                icon: 'securityApp',
-                path: '/app/security-configuration',
-                showOnHomePage: true,
-                category: FeatureCatalogueCategory.ADMIN
-            };
-        });
+
+        if(systemstate.getRestApiInfo().has_api_access === true) {
+
+            chrome.getNavLinkById("security-configuration").hidden = false;
+            FeatureCatalogueRegistryProvider.register(() => {
+                return {
+                    id: 'security-configuration',
+                    title: 'Security Configuration',
+                    description: 'Configure users, roles and permissions for Open Distro Security.',
+                    icon: 'securityApp',
+                    path: '/app/security-configuration',
+                    showOnHomePage: true,
+                    category: FeatureCatalogueCategory.ADMIN
+                };
+            });
+        } else {
+            return;
+        }
     });
 }
`

Any thoughts ???

**Comment by hardik-k-shah** [comment]
@tremmert  I agree with you. It should not be visible. We also want to add ability for user to change their own password. ([issue#47](https://github.com/opendistro-for-elasticsearch/security/issues/47))  But it can go into different tab/link.  @elfisher @aetter Are you agree with this?

**Comment by tremmert** [comment]
Any view on this: @elfisher @aetter ? Thank you !

**Comment by aetter** [comment]
Yep, we should not show the **Security** tab unless the user can do something there. Viewing the **Authentication and Authorization** link doesn't count as "doing something." :)

Changing your own password would qualify, but like @hardik-k-shah said, maybe that should be a separate **Profile** or **Account** tab. I don't have a strong opinion either way. @stevensideyliu or @elfisher might.

Side note: logging in with no permissions other than `kibana_user` and clicking on **Authentication and Authorization** actually logs me out, so that's another unfortunate user experience that we could eliminate by locking down access to the tab.

**Comment by stevensideyliu** [comment]
Agree with removing this tab if the account does not have permission to use. 
The second suggestion needs more investigation on design. We should seporate the two issues and find clear resolutions for each. 

We should work with @tremmert on this suggestion. Seems like a small change with great UX impact.

**Comment by davidlago** [comment]
We are doing some "spring cleaning in the fall", and to make sure we focus our energies on the right issues and we get a better picture of the state of the repo, we are closing all issues that we are carrying over from the ODFE era (ODFE is no longer supported/maintained, see post [here](https://opendistro.github.io/for-elasticsearch/blog/2022/08/Open-Distro-is-archived/)).

If you believe this issue should still be considered for current versions of OpenSearch, apologies! Please let us know by re-opening it.

Thanks!
